src/ml/neural_net/mps_layer_instance_norm_data_loader.mm

```diff
@@ -237,30 +199,30 @@ - (MPSCNNNormalizationGammaAndBetaState *)updateGammaAndBetaWithCommandBuffer:(i
       [_adamBeta encodeToCommandBuffer:commandBuffer
                    inputGradientVector:gradientBiasesVector
                      inputValuesVector:inputBiasesVector
-                   inputMomentumVector:_betaMomentumVectorArray[_currentStyle]
-                   inputVelocityVector:_betaVelocityVectorArray[_currentStyle]
-                    resultValuesVector:_betaVectorArray[_currentStyle]];
+                   inputMomentumVector:[[style_props objectAtIndex: _currentStyle] betaMomentumVector] 
+                   inputVelocityVector:[[style_props objectAtIndex: _currentStyle] betaVelocityVector]
+                    resultValuesVector:[[style_props objectAtIndex: _currentStyle] betaVector]];
 
   }
 
-    return stateArray[_currentStyle];
+    return [[style_props objectAtIndex: _currentStyle] state];
 }
 
 - (void)checkpointWithCommandQueue:(nonnull id<MTLCommandQueue>)commandQueue {
   id<MTLCommandBuffer> commandBuffer = [commandQueue commandBuffer];
   id<MTLBlitCommandEncoder> blit = commandBuffer.blitCommandEncoder;
 
-  for (size_t index = 0; index < _styles; index ++){
-    [blit synchronizeResource:_betaMomentumBufferArray[index]];
-    [blit synchronizeResource:_betaVelocityBufferArray[index]];
-    [blit synchronizeResource:nonnull_cast(stateArray[index].beta)];
+  for (size_t index = 0; index < _styles; index ++) {
+    [blit synchronizeResource:[[style_props objectAtIndex: index] betaMomentumBuffer]];
```

Comment:
Let's factor out the repeated array indexing to get the props for this style once, at the top of the loop!

Reply:
Looks good?!